src/java_bytecode/library/src/java/lang/Thread.java

```diff
@@ -0,0 +1,1994 @@
+  // }
+
+  /**
+   * Throws CloneNotSupportedException as a Thread can not be meaningfully
+   * cloned. Construct a new Thread instead.
+   *
+   * @throws  CloneNotSupportedException
+   *          always
+   */
+  //@Override
+  protected Object clone() throws CloneNotSupportedException {
+    throw new CloneNotSupportedException();
+  }
+
+  /**
+   * Allocates a new {@code Thread} object. This constructor has the same
+   * effect as {@linkplain #Thread(ThreadGroup,Runnable,String) Thread}
+   * {@code (null, null, gname)}, where {@code gname} is a newly generated
+   * name. Automatically generated names are of the form
+   * {@code "Thread-"+}<i>n</i>, where <i>n</i> is an integer.
+   */
+  public Thread()
+  {
+    target = this;
+    tid = nextThreadID();
+  }
+
+  /**
+   * Allocates a new {@code Thread} object. This constructor has the same
+   * effect as {@linkplain #Thread(ThreadGroup,Runnable,String) Thread}
+   * {@code (null, target, gname)}, where {@code gname} is a newly generated
+   * name. Automatically generated names are of the form
+   * {@code "Thread-"+}<i>n</i>, where <i>n</i> is an integer.
+   *
+   * @param  target
+   *         the object whose {@code run} method is invoked when this thread
+   *         is started. If {@code null}, this classes {@code run} method does
+   *         nothing.
+   */
+  public Thread(Runnable target)
+  {
+    this.target = target;
+    tid = nextThreadID();
+  }
+
+  // /**
+  //  * Creates a new Thread that inherits the given AccessControlContext.
+  //  * This is not a public constructor.
+  //  */
+  // Thread(Runnable target, AccessControlContext acc) {
+  //   init(null, target, "Thread-" + nextThreadNum(), 0, acc, false);
+  // }
+
+  // /**
+   // * Allocates a new {@code Thread} object. This constructor has the same
+   // * effect as {@linkplain #Thread(ThreadGroup,Runnable,String) Thread}
+   // * {@code (group, target, gname)} ,where {@code gname} is a newly generated
+   // * name. Automatically generated names are of the form
+   // * {@code "Thread-"+}<i>n</i>, where <i>n</i> is an integer.
+   // *
+   // * @param  group
+   // *         the thread group. If {@code null} and there is a security
+   // *         manager, the group is determined by {@linkplain
+   // *         SecurityManager#getThreadGroup SecurityManager.getThreadGroup()}.
+   // *         If there is not a security manager or {@code
+   // *         SecurityManager.getThreadGroup()} returns {@code null}, the group
+   // *         is set to the current thread's thread group.
+   // *
+   // * @param  target
+   // *         the object whose {@code run} method is invoked when this thread
+   // *         is started. If {@code null}, this thread's run method is invoked.
+   // *
+   // * @throws  SecurityException
+   // *          if the current thread cannot create a thread in the specified
+   // *          thread group
+   // */
+  // public Thread(ThreadGroup group, Runnable target) {
```

Comment:
The strategy in models-library is to not remove any public methods, as to not change the interface of the JDK. Instead, we comment out the body and replace it with a `CProver.notModelled()` statement. Would you consider doing something similar?